tcgaImport.py

```diff
@@ -520,52 +525,67 @@ def fileScan(self, path, dataSubType):
         target = None
         colName = None
         colType = None
-        for line in iHandle:
-            if colName is None:
-                colName = line.rstrip().split("\t")                     
-                if colName[0] == "Hybridization REF" or colName[0] == "Sample REF":
-                    mode=2
-                elif colName[0] == "Chromosome" or colName[0] == "chromosome":
-                    mode=1
-                    target=os.path.basename( path ).split('.')[0] #seg files are named by the filename before the '.' extention
-                elif colName[1] == "chrom":
-                    mode = 3
-                    target=os.path.basename( path ).split('.')[0] #seg files are named by the filename before the '.' extention
-                    
-                for i in range(len(colName)):
-                    if commonMap.has_key( colName[i] ):
-                        colName[i] = commonMap[ colName[i] ]
-            elif mode==2 and colType is None:
-                colType=line.rstrip().split("\t")
-                for i in range(len(colType)):
-                    if commonMap.has_key( colType[i] ):
-                        colType[i] = commonMap[ colType[i] ]
+        firstLine = iHandle.readline()
+        colName = firstLine.rstrip().split("\t")
+        if colName[0] == "Hybridization REF" or colName[0] == "Sample REF":
+            mode = 2
+        elif colName[0] == "Chromosome"  or colName[0] == "chromosome":
+            mode = 1
+            target = os.path.basename( path ).split('.')[0]
+        elif colName[1] == "chrom":
+            mode = 3
+            target = os.path.basename( path ).split('.')[0]
+
+        def addTargetColumn(df, target):
+            tmp = df
+            tmp["target"] = target
+            return tmp
+        
+        
+        if mode == 2:
+            colName = [correctKey(colName[i]) for i in range(len(colName))]
+            secondLine = iHandle.readline()
+            colType = secondLine.rstrip().split("\t")
+            colType = [correctKey(colType[i]) for i in range(len(colType))]
+            tmp = pd.read_csv(iHandle, sep="\t", header=None, names=colType)
+            for i in range(1, len(colType)):
```

Comment:
idx = [col in self.dataSubTypes[dataSubType]['probeFields'] in for col  in colType]
tmp = tmp[idx]